app/src/main/java/org/oppia/app/drawer/NavigationDrawerFragmentPresenter.kt

```diff
@@ -110,11 +111,9 @@ class NavigationDrawerFragmentPresenter @Inject constructor(
               activity,
               internalProfileId
             )
+          intent.setFlags(FLAG_ACTIVITY_SINGLE_TOP)
           fragment.activity!!.startActivity(intent)
-          if (previousMenuItemId != null &&
-            NavigationDrawerItem.valueFromNavId(previousMenuItemId!!) !=
```

Comment:
### Why this causes the crash

As we know that the `AdministratorControlsActivity` is not a part of navigation menu item, rather it's a different layout inside the navigation drawer. So, when we click on the `AdministratorControls` second time, `AdministratorControlsActivityPresenter` calls the `setUpNavigationDrawer` and pass `0` as a menu item which is not a menuItem and calling `valueFromNavId ` for `0` brings crash.

Reply:
###  why this fixes the underlying crash

What I fix here is:
1. added a flag to intent to the activity created only once and used is already existed, as for other menu item this case is not applicable, as those menu item just close the nav drawer only if already open 
`if (previousMenuItemId != menuItemId) then close the drawer`
2. Secondly, I had update the if condition `if (previousMenuItemId != NavigationDrawerItem.HOME.ordinal)` then finish the last activity.  

There is a bug with the nav drawer flow in between menu and admincontrol which can solved with this issue - https://github.com/oppia/oppia-android/issues/1762

Reply:
Thanks, can you also specify these details in the opening comment in the PR & in-code? In general, these are the two places most likely to be looked at in the future when trying to figure out what's going on (either the purpose of the PR for the former, or the purpose of the code for the latter).